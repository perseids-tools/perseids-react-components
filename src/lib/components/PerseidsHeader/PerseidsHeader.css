.perseids-react-components--navbar {
  min-height: 72px;
}

.perseids-react-components--navbar-brand {
  padding-top: 8px;
  padding-bottom: 8px;
}

.perseids-react-components--navbar-logo-img {
  height: 40px;
}

.navbar-light .navbar-nav .nav-link {
  color: rgba(0, 0, 0, 0.54);
}

.navbar-light .navbar-text {
  color: rgba(0, 0, 0, 0.54);
}
