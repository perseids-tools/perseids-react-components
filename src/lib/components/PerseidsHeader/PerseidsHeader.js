import React from 'react';
import PropTypes from 'prop-types';

import './PerseidsHeader.css';

import perseidsLogoTransparent from './perseids-logo-transparent.png';

const PerseidsHeader = ({ children }) => (
  <header className="navbar navbar-expand-md navbar-light bg-light perseids-react-components--navbar">
    <a className="navbar-brand" href="https://www.perseids.org">
      <img className="perseids-react-components--navbar-logo-img" src={perseidsLogoTransparent} title="perseids logo" alt="perseids logo" />
    </a>
    {children}
  </header>
);

PerseidsHeader.defaultProps = {
  children: false,
};

PerseidsHeader.propTypes = {
  children: PropTypes.element,
};

export default PerseidsHeader;
