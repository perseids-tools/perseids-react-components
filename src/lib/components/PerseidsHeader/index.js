import PerseidsHeader from './PerseidsHeader';

export default PerseidsHeader;
