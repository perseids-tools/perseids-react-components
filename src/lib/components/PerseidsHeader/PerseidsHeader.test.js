import React from 'react';
import renderer from 'react-test-renderer';

import PerseidsHeader from './PerseidsHeader';

it('renders with a link to perseids.org', () => {
  const tree = renderer.create(<PerseidsHeader />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render a different element', () => {
  const component = (
    <PerseidsHeader component="div" />
  );

  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render a custom component', () => {
  const MyComponent = ({ className, name, children }) => (
    <span className={className} name={name}>
      {children}
    </span>
  );
  const component = (
    <PerseidsHeader component={MyComponent} props={{ name: 'test' }} />
  );

  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});

it('can render children between copyright notice and icons', () => {
  const component = (
    <PerseidsHeader>
      Header text
    </PerseidsHeader>
  );
  const tree = renderer.create(component).toJSON();

  expect(tree).toMatchSnapshot();
});
