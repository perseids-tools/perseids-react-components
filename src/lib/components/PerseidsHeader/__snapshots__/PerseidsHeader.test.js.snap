// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render children between copyright notice and icons 1`] = `
<header
  className="navbar navbar-expand-md navbar-light bg-light perseids-react-components--navbar"
>
  <a
    className="navbar-brand perseids-react-components--navbar-brand"
    href="https://www.perseids.org"
  >
    <img
      alt="perseids logo"
      className="perseids-react-components--navbar-logo-img"
      src="perseids-logo-transparent.png"
      title="perseids logo"
    />
  </a>
  Header text
</header>
`;

exports[`renders with a link to perseids.org 1`] = `
<header
  className="navbar navbar-expand-md navbar-light bg-light perseids-react-components--navbar"
>
  <a
    className="navbar-brand perseids-react-components--navbar-brand"
    href="https://www.perseids.org"
  >
    <img
      alt="perseids logo"
      className="perseids-react-components--navbar-logo-img"
      src="perseids-logo-transparent.png"
      title="perseids logo"
    />
  </a>
</header>
`;
